src/railfence.rs

```diff
@@ -0,0 +1,284 @@
+//! The Railfence Cipher is a transposition cipher. It has a very low keyspace and is therefore
+//!incredibly insecure.
+//!
+//! This implementation currently transposes all input characters including whitespace and
+//!punctuation.
+
+/// A Railfence cipher.
+///
+/// This struct is created by the `new()` method. See its documentation for more.
+pub struct Railfence {
+    key: usize,
+}
+
+impl Railfence {
+    /// Initialise a Railfence cipher given a specific key.
+    ///
+    /// Will return `Err` if the key is zero.
+    pub fn new(key: usize) -> Result<Railfence, &'static str> {
+        if key == 0 {
+            return Err("Invalid key. Railfence key cannot be zero.");
+        }
+
+        Ok(Railfence { key: key })
+    }
+
+    /// Encrypt a message using a Railfence cipher.
+    ///
+    /// # Examples
+    /// Basic usage:
+    ///
+    /// ```
+    /// use cipher_crypt::railfence::Railfence;
+    ///
+    /// let r = Railfence::new(3).unwrap();
+    /// assert_eq!("Src s!ue-ertmsaepseeg", r.encrypt("Super-secret message!"));
+    /// ```
+    pub fn encrypt(&self, message: &str) -> String {
+        // Encryption process:
+        //   First a table is created with a height given by the key and a length
+        //   given by the message length.
+        //   e.g.
+        //   For a key of 3 and the message "Hello, World!" of length 13:
+        //      .............
+        //      .............
+        //      .............
+        //   The message can then be written onto the grid in a zigzag going right:
+        //      H...o...o...!
+        //      .e.l.,.W.r.d.
+        //      ..l... ...l..
+        //   The encrypted message is then read line by line:
+        //      Hoo!el,Wrdl l 
+        
+        
+        // A key of one causes a problem when calculating the cycle later on,
```

Comment:
I will make it more explicit, but a key of one actually does cause an error if this 'if' statement is removed. Its because the expression for the cycle evaluates to zero, and you then do a modulus operation with it.